judge/admin/contest.py

```diff
@@ -219,6 +223,11 @@ def get_form(self, *args, **kwargs):
         ).distinct()
         return form
 
+    def save_model(self, request, obj, form, change):
+        super(ContestAdmin, self).save_model(request, obj, form, change)
+        if form.changed_data and any(f in form.changed_data for f in ('format_config', 'format_name')):
```

Comment:
what about when problems or problem weights change? does anything need to be updated then?